cpp/include/cudf/types.h

```diff
@@ -85,6 +85,36 @@ typedef struct {
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+/**---------------------------------------------------------------------------*
+ * @union gdf_data
+ * @brief Union used for scalar type.
+ * It stores a unique value for scalar type.
+ * It has a direct relationship with the gdf_dtype.
+ *---------------------------------------------------------------------------**/
+// TODO: #1119 Use traits to set `gdf_data` elements
+typedef union {
+  int8_t   si08;  /**< GDF_INT8      */
+  int16_t  si16;  /**< GDF_INT16     */
+  int32_t  si32;  /**< GDF_INT32     */
+  int64_t  si64;  /**< GDF_INT64     */
+  float    fp32;  /**< GDF_FLOAT32   */
+  double   fp64;  /**< GDF_FLOAT64   */
+  int32_t  dt32;  /**< GDF_DATE32    */
```

Comment:
We can't use the `wrapper` types because it will break CFFI, but we can at least use the `typedef`s.

```suggestion
  gdf_date32  dt32;  /**< GDF_DATE32    */
```

Reply:
Why the shorthands? `data_in_32_bits` sounds much better than the cryptic `dt32`.